cni/network/invoker_cns.go

```diff
@@ -356,3 +304,158 @@ func (invoker *CNSIPAMInvoker) Delete(address *net.IPNet, nwCfg *cni.NetworkConf
 
 	return nil
 }
+
+func getRoutes(cnsRoutes []cns.Route) ([]*cniTypes.Route, error) {
+	routes := make([]*cniTypes.Route, 0)
+	for _, route := range cnsRoutes {
+		_, dst, routeErr := net.ParseCIDR(route.IPAddress)
+		if routeErr != nil {
+			return nil, fmt.Errorf("unable to parse destination %s: %w", route.IPAddress, routeErr)
+		}
+
+		routes = append(routes,
+			&cniTypes.Route{
+				Dst: *dst,
+				GW:  net.ParseIP(route.GatewayIPAddress),
+			})
+	}
+
+	return routes, nil
+}
+
+func configureDefaultAddResult(info *IPResultInfo, addConfig *IPAMAddConfig, addResult *IPAMAddResult, overlayMode bool) error {
+	// set the NC Primary IP in options
+	// SNATIPKey is not set for ipv6
+	if net.ParseIP(info.ncPrimaryIP).To4() != nil {
+		addConfig.options[network.SNATIPKey] = info.ncPrimaryIP
+	}
+
+	ip, ncIPNet, err := net.ParseCIDR(info.podIPAddress + "/" + fmt.Sprint(info.ncSubnetPrefix))
+	if ip == nil {
+		return errors.Wrap(err, "Unable to parse IP from response: "+info.podIPAddress+" with err %w")
+	}
+
+	ncgw := net.ParseIP(info.ncGatewayIPAddress)
+	if ncgw == nil {
+		// TODO: Remove v4overlay and dualstackoverlay options, after 'overlay' rolls out in AKS-RP
+		if !overlayMode {
+			return errors.Wrap(errInvalidArgs, "%w: Gateway address "+info.ncGatewayIPAddress+" from response is invalid")
+		}
+
+		if net.ParseIP(info.podIPAddress).To4() != nil { //nolint:gocritic
+			ncgw, err = getOverlayGateway(ncIPNet)
+			if err != nil {
+				return err
+			}
+		} else if net.ParseIP(info.podIPAddress).To16() != nil {
+			ncgw = net.ParseIP(overlayGatewayV6IP)
+		} else {
+			return errors.Wrap(err, "No podIPAddress is found: %w")
+		}
+	}
+
+	// construct ipnet for result
+	resultIPnet := net.IPNet{
+		IP:   ip,
+		Mask: ncIPNet.Mask,
+	}
+
+	if ip := net.ParseIP(info.podIPAddress); ip != nil {
+		defaultInterfaceInfo := addResult.defaultInterfaceInfo.ipResult
+		if defaultInterfaceInfo == nil {
+			defaultInterfaceInfo = &cniTypesCurr.Result{}
```

Comment:
in which case this will happen? looks to me over cautious.. if its not gonna happen, i would suggest to remove this check

Reply:
this will be nil until we handle the default interface (call configureDefaultAddResult

Reply:
can `configureDefaultAddResult` be called with `addResult.defaultInterfaceInfo.ipResult` as nil?

Reply:
yes, the first call to configureDefaultAddResult will have nil addResult.defaultInterfaceInfo.ipResult

Reply:
why so? If thats the case, can this check be even before this function called?